src/erlcloud_kinesis.erl

```diff
@@ -560,3 +565,135 @@ split_shards(StreamName, ShardToSplit, NewStartingHashKey) ->
+%% @doc
+%% Kinesis API:
+%% [http://docs.aws.amazon.com/kinesis/latest/APIReference/API_ListTagsForStream.html]
+%%
+%% Lists the tags for the specified Amazon Kinesis stream.
+%%
+%% erlcloud_kinesis:list_tags_for_stream(<<"stream_name">>, <<"key2">>, 2).
+%% {ok, [
+%%       {<<"HasMoreTags">>, true},
+%%       {<<"Tags">>,        [[{<<"Key">>, <<"k1">>}, {<<"Value">>, <<"v1">>}],
+%%                            [{<<"Key">>, <<"k2">>}, {<<"Value">>, <<"v2">>}]]}
+%%      ]
+%% }.
+%%
+%% @end
+%%------------------------------------------------------------------------------
+-spec list_tags_for_stream(binary()) -> {ok, proplist()} | {error, any()}.
+list_tags_for_stream(StreamName) ->
+    list_tags_for_stream(StreamName, default_config()).
+
+-spec list_tags_for_stream(binary(), aws_config()) ->
+    {ok, proplist()} | {error, any()}.
+list_tags_for_stream(StreamName, Config) when is_record(Config, aws_config) ->
+    list_tags_for_stream(StreamName, undefined, undefined, Config).
+
+-spec list_tags_for_stream(binary(),
+                           binary() | undefined,
+                           integer() | undefined) ->
+    {ok, proplist()} | {error, any()}.
+list_tags_for_stream(StreamName, ESK, Limit) ->
+    list_tags_for_stream(StreamName, ESK, Limit, default_config()).
+
+-spec list_tags_for_stream(binary(),
+                           binary() | undefined,
+                           integer() | undefined,
+                           aws_config()) ->
+    {ok, proplist()} | {error, any()}.
+list_tags_for_stream(StreamName, ESK, Limit, Config)
+  when is_record(Config, aws_config) ->
+    Json      = [{<<"StreamName">>,           StreamName},
+                 {<<"ExclusiveStartTagKey">>, ESK},
+                 {<<"Limit">>,                Limit}],
+    JsonUpd   = lists:filter(fun filter_undefined_params/1, Json),
+    Operation = "Kinesis_20131202.ListTagsForStream",
+    erlcloud_kinesis_impl:request(Config, Operation, JsonUpd).
+
+filter_undefined_params({_, undefined}) -> false;
+filter_undefined_params(_)              -> true.
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% Wrapper around 'list_tags_for_stream' to return {ok, <list_of_all_tags>}.
+%%
+%% erlcloud_kinesis:list_all_tags_for_stream(<<"stream_name">>).
+%% {ok, [{<<"k1">>, <<"v1">>},
+%%       {<<"k2">>, <<"v2">>}]
+%% }.
+%%
+%% @end
+%%------------------------------------------------------------------------------
+-spec list_all_tags_for_stream(binary()) -> {ok, proplist()} | {error, any()}.
+list_all_tags_for_stream(StreamName) ->
+    list_all_tags_for_stream(StreamName, default_config()).
+
+-spec list_all_tags_for_stream(binary(), aws_config()) ->
+    {ok, proplist()} | {error, any()}.
+list_all_tags_for_stream(StreamName, Config) ->
+    case list_tags_for_stream(StreamName, Config) of
+        {ok, [_HasMoreTags, {<<"Tags">>, Tags}]} ->
```

Comment:
we probably shouldn't depend on specific structure of aws response here, they may add fields in future